src/predict_kfold.py

```
@@ -0,0 +1,123 @@
+
+    print(f"Starting prediction on {len(test_filenames)} images...")
+    final_predictions = []
+
+    # 3. Iterate over images
+    for image_name in tqdm(test_filenames, desc="K-Fold Ensemble"):
+        image_path = os.path.join(TEST_IMAGE_DIR, image_name)
+
+        try:
+            # Load original image
+            img_original = Image.open(image_path).convert('L')
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)